plugins/module_utils/system_access_users_utils.py

```diff
@@ -39,70 +39,114 @@
     xml_utils,
     opnsense_utils,
 )
+
 from ansible_collections.puzzle.opnsense.plugins.module_utils.config_utils import (
     OPNsenseModuleConfig,
 )
-from ansible_collections.puzzle.opnsense.plugins.module_utils.enum_utils import ListEnum
 
 
-class OPNSenseGroupNotFoundError(Exception):
+class OPNSenseCryptReturnError(Exception):
     """
-    Exception raised when an OPNsense group is not found.
+    Exception raised when the return value of the instance is not what is expected
     """
 
 
-class OPNSenseNotValidBase64APIKeyError(Exception):
+class OPNSenseGroupNotFoundError(Exception):
     """
-    Exception raised when a not valid base32 api code is provided
+    Exception raised when an OPNsense group is not found.
     """
 
 
-class OPNSenseCryptReturnError(Exception):
+class OPNSenseHashVerifyReturnError(Exception):
     """
     Exception raised when the return value of the instance is not what is expected
     """
 
 
-class OPNSensePasswordVerifyReturnError(Exception):
+class OPNSenseNotValidBase64APIKeyError(Exception):
     """
-    Exception raised when the return value of the instance is not what is expected
+    Exception raised when a not valid base32 api code is provided
     """
 
 
-def password_verify(existing_user_password: str, password: Optional[str]) -> bool:
+def hash_verify(existing_hashed_string: str, plain_string: Optional[str]) -> bool:
     """
-    Verify if provided password matches the stored password using OPNsense's PHP command.
+    Verifies if a plain string matches an existing hashed string.
 
     Args:
-        existing_user_password (str): The hashed password stored in the XML config.
-        password (str): The plaintext password to verify.
+        existing_hashed_string (str): The existing hashed string to verify against.
+        plain_string (Optional[str]): The plain string to verify.
 
     Returns:
-        bool: True if passwords match, False otherwise.
+        bool: True if the plain string matches the hashed string, otherwise False.
 
     Raises:
-        OPNSensePasswordVerifyReturnError: If an error occurs during verification.
+        OPNSenseHashVerifyReturnError: If an error occurs during hash verification.
```

Comment:
```suggestion
        OPNsenseHashVerifyReturnError: If an error occurs during hash verification.
```

Reply:
Thanks for the suggestion, fixed in cfcd2b4.